source/org/zfin/figure/service/ImageThumbnailAndMediumSwapTask.java

```diff
@@ -0,0 +1,178 @@
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+public class ImageThumbnailAndMediumSwapTask extends AbstractScriptWrapper {
+    private boolean dryRun = true;
+    private List<String> restrictedPublications;
+
+    public static void main(String[] args) throws IOException {
+        ImageThumbnailAndMediumSwapTask task = new ImageThumbnailAndMediumSwapTask();
+        task.runTask();
+        System.exit(0);
+    }
+
+    public void runTask() throws IOException {
+        initAll();
+        initConfig();
+
+
+        FigureRepository figureRepository = RepositoryFactory.getFigureRepository();
+        List<Image> images = figureRepository.getAllImagesWithFigures();
+        LOG.info("Found " + images.size() + " images with figures.");
+        if (!restrictedPublications.isEmpty()) {
+            images.removeIf(image -> !restrictedPublications.contains(image.getFigure() == null ? "" : image.getFigure().getPublication().getZdbID()));
+            LOG.info("Found " + images.size() + " images after filtering by publication ID.");
+        }
+
+        runFixesForThumbnailsAndMediums(images);
+    }
+
+    private void initConfig() {
+        boolean runFixes = System.getenv().getOrDefault("RUN_IMAGE_FIXES", "false").equalsIgnoreCase("true");
+        dryRun = !runFixes;
+        String restrictedPublicationsString = System.getenv("FIX_PUB_IDS");
+        if (StringUtils.isNotEmpty(restrictedPublicationsString)) {
+            restrictedPublications = List.of(restrictedPublicationsString.split(","));
+        } else {
+            restrictedPublications = Collections.emptyList();
+        }
+
+        if (dryRun) {
+            System.out.println("Dry run.  No changes will be made. Set RUN_IMAGE_FIXES environment variable, or runImageFixes property to run.");
+        } else {
+            System.out.println("Executing changes with imagemagick.");
+        }
+
+        if (!restrictedPublications.isEmpty()) {
+            System.out.println("Only fixing images for publications: " + restrictedPublications);
+        } else {
+            System.out.println("Fixing images for all publications. To restrict, set FIX_PUB_IDS environment variable to comma separated list of IDs.");
+        }
+    }
+
+    private void runFixesForThumbnailsAndMediums(List<Image> images) throws IOException {
+        for (Image image : images) {
+            String thumbnail = image.getThumbnail();
+            String medium = image.getMedium();
+            String fullSizeImagePath = ZfinPropertiesEnum.LOADUP_FULL_PATH + File.separator + image.getImageFilename();
+            File thumbnailFile = new File(ZfinPropertiesEnum.LOADUP_FULL_PATH + File.separator + thumbnail);
+            File mediumFile = new File(ZfinPropertiesEnum.LOADUP_FULL_PATH + File.separator + medium);
+
+            if (!thumbnailFile.exists()) {
```

Comment:
Could be a single if-clause with two arguments...

Reply:
Thanks, fixed